develop/entities/first-entity.md

```diff
@@ -0,0 +1,163 @@
+
+Each part is added with a offset point for proper animation and alignment, ensuring the model appears correctly in-game.
+
+::: info
+The higher Y values in the model, the lower you are in the entity. This is the reverse compared to in-game coordinates.
+:::
+
+We'll now need to create a `ModEntityModelLayers` class in the client package. This entity only has a single texture layer, but other entities may use multiple - think of the secondary skin layer on mobs like the `Player` or a `Spider`'s eyes.
+
+@[code transcludeWith=:::model_layer](@/reference/latest/src/client/java/com/example/docs/entity/model/ModEntityModelLayers.java)
+
+This class must then be initialized in the mod's client initializer.
+
+@[code transcludeWith=::register_client](@/reference/latest/src/client/java/com/example/docs/entity/ExampleModCustomEntityClient.java)
+
+### Setting up the Texture {#setting-up-texture}
+
+::: warning
+The size of the texture should match the values in the `LayerDefinition.create(modelData, 64, 32);`, 64 pixels wide and 32 pixels tall.
+:::
+
+Each model part / box is expecting a net on the texture in a particular location. By default, it's expecting it at `0, 0` (the top left), but this can be changed by calling the texOffs function in CubeListBuilder.
+
+For example purposes, you can use this example texture for `assets/example-mod/textures/entity/mini_golem.png`
+
+<DownloadEntry visualURL="/assets/develop/entity/mini_golem.png" downloadURL="/assets/develop/entity/mini_golem_small.png">Texture</DownloadEntry>
+
+### Creating the Renderer {#creating-the-renderer}
+
+A mob's entity renderer enables you to view your entity in-game. We'll create a new class, `MiniGolemEntityRenderer`, which will tell Minecraft what texture, model and entity render state to use for this entity.
+
+@[code transcludeWith=:::renderer](@/reference/latest/src/client/java/com/example/docs/entity/renderer/MiniGolemEntityRenderer.java)
+
+This is also where the shadow radius is set, for this entity that will be `0.375f`.
+
+### Adding Walking Animations {#walking-animations}
+
+The following code can be added to the `MiniGolemEntityModel` class to give the entity a walking animation.
+
+@[code transcludeWith=:::model_animation](@/reference/latest/src/client/java/com/example/docs/entity/model/MiniGolemEntityModel.java)
+
+Most of the math is provided by Blockbench.
+
+Looking into the game, you now have all you need to spawn the entity with `/summon example-mod:mini_golem`!
+
+![Spawn Egg showcase](/assets/develop/entity/mini_golem_summoned.png)
+
+## Adding a Dancing Animation {#adding-a-dancing-animation}
+
+Let's make this entity a bit more lively by giving it a dancing animation. We'll create a `MiniGolemAnimations` class with a simple animation that allows the Mini Golem to dance.
+
+@[code transcludeWith=:::dancing_animation](@/reference/latest/src/client/java/com/example/docs/entity/animation/MiniGolemAnimations.java)
```

Comment:
As explained in the big long comment, I think this part should go in its own section at the end (just before the spawn egg).